lib/dialects/postgres/data-types.js

```diff
@@ -52,24 +52,24 @@ DATE.prototype.toSql = function() {
 
 var INTEGER = function() {
   if (!(this instanceof INTEGER)) return new INTEGER();
-  BaseTypes.INTEGER.apply(this, arguments);
-
-  // POSTGRES does not support length on INTEGER
-  delete this._length;
-  delete this.options.length;
```

Comment:
Ah because you are removing the super call, i'm not sure that's the best way to go, it'd be too easy to add logic to the super and then forget to copy it down here.